crates/router/src/services/api/client.rs

```diff
@@ -218,68 +179,41 @@ where
     ) -> CustomResult<reqwest::Response, ApiClientError>;
 
     fn add_request_id(&mut self, request_id: RequestId);
+
     fn get_request_id(&self) -> Option<String>;
+
     fn add_flow_name(&mut self, flow_name: String);
 }
 
 dyn_clone::clone_trait_object!(ApiClient);
 
 #[derive(Clone)]
 pub struct ProxyClient {
-    proxy_client: reqwest::Client,
-    non_proxy_client: reqwest::Client,
-    whitelisted_urls: Vec<String>,
+    proxy_config: Proxy,
```

Comment:
Added the `proxy_config` field, which is required when constructing a client where certificates are specified.